firedrake/preconditioners/patch.py

```diff
@@ -568,11 +571,41 @@ def select_entity(p, dm=None, exclude=None):
 class PlaneSmoother(object):
     @staticmethod
     def coords(dm, p):
-        coordsSection = dm.getCoordinateSection()
-        coordsDM = dm.getCoordinateDM()
-        dim = coordsDM.getDimension()
-        coordsVec = dm.getCoordinatesLocal()
-        return dm.getVecClosure(coordsSection, coordsVec, p).reshape(-1, dim).mean(axis=0)
+        mesh = dm.getAttr("__firedrake_mesh__")
+
+        ele = mesh.coordinates.function_space().ufl_element()
+        assert ele.family() in ["Lagrange", "Discontinuous Lagrange"]
+        if ele.family() == "Discontinuous Lagrange":
+            # Ah, we got a periodic mesh. We need to interpolate to CGk
+            # with access descriptor MAX to define a consistent opinion
+            # about where the vertices are.
+
+            coords = dm.getAttr("__firedrake_interpolated_coords__")
+            if coords is None:
+                CGkele = ele.reconstruct(family="Lagrange")
+                # Can't do
+                # CGk = FunctionSpace(mesh, CGkele)
+                # because equality fails between a mesh and a weakref.proxy.
```

Comment:
I think this comment is misleading:
```
mesh = ...
mesh2 = weakref.proxy(mesh)
mesh2 == mesh
=> True
``` 
(at least for me)

But weakref.proxies are not hashable, so the interpolation fails for me. I think.

Reply:
Thanks, fixed.